apis/v1beta1/collector_webhook.go

```diff
@@ -100,6 +101,9 @@ func (c CollectorWebhook) Default(_ context.Context, obj runtime.Object) error {
 	if len(otelcol.Spec.ManagementState) == 0 {
 		otelcol.Spec.ManagementState = ManagementStateManaged
 	}
+	if c.disableDefaulting {
+		return nil
+	}
```

Comment:
Do we actually need this?
From my point of view we should limit the defaulting only to components we actually configured a parser for.
In that case, we use `0.0.0.0` to default if at least a parser+port is configured.

Reply:
i think so, i believe there will be users that plainly don't want us touching their config at all. I can imagine a scenario where a user has a custom receiver which keeps getting an endpoint to set to 0.0.0.0 which doesn't allow for setting to that host or doesn't have an endpoint... though maybe to @swiatekm's point we should just make this a featuregate and eventually enable it for everyone. I do think that makes sense too... I'm going to do that instead.

Reply:
When we limit it to known parsers, it should not be an issue if a collector contains custom components.
Regarding the general modification I would not worry to much. We also do other defaulting and so on.

In case you want to run the collector only on localhost (default behaviour),  you still can configure the endpoint explicit.

If we still want to add an option, I would prefer something to overwrite the default value. E.g.  replace `0.0.0.0` with `$POD_IP` or `localhost`.